src/plugins/workspace/public/components/workspace_form/fields/workspace_name_field.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiCompressedFieldText, EuiCompressedFormRow } from '@elastic/eui';
+import { i18n } from '@osd/i18n';
+import React, { useCallback } from 'react';
+
+import { MAX_WORKSPACE_NAME_LENGTH } from '../../../../common/constants';
+
+export interface WorkspaceNameFieldProps {
+  value?: string;
+  onChange: (newValue: string) => void;
+  error?: string;
+  readOnly?: boolean;
+}
+
+export const WorkspaceNameField = ({
+  value,
+  error,
+  readOnly,
+  onChange,
+}: WorkspaceNameFieldProps) => {
+  const handleChange = useCallback(
```

Comment:
Shall we allow user to keep editing but display error message when exceed limit?

Reply:
I'm open to this. Can be discuss with designer about that behavior.

Reply:
I've updated this part according the latest feedback. Could you help me take a look?